src/main/java/graphql/execution/AbortExecutionException.java

```diff
@@ -2,7 +2,7 @@
 
```

Comment:
This one feels like a GraphqlInstanceException no???

Reply:
`AbortExecutionException` is being raised [when the query complexity has reached the limit](https://github.com/graphql-java/graphql-java/blob/3c44f065a902f8318811f2ad240af372df68e602/src/main/java/graphql/analysis/MaxQueryComplexityInstrumentation.java#L73) so in this case it should fall under `GraphQLQueryException`. It's also being raised if there are errors [when validating the query fields](https://github.com/graphql-java/graphql-java/blob/688d15769e627c6fb3e55e2efd1ad2fe41b86626/src/main/java/graphql/execution/instrumentation/fieldvalidation/FieldValidationInstrumentation.java#L43).

Reply:
This ones weird because there is a long standing need to abort the "query" before execution (like these complexity cases) and an unsatisfied need to let a data fetcher abort the query which we don't have today but would like

So as it stands today the hierarchy is right..but I am thinking into the future on whether we want to reserve this name for an as yet unused feature

Reply:
Can you elaborate on this: `an unsatisfied need to let a data fetcher abort the query which we don't have today but would like`
What are the reasons the data fetcher would abort the query? Can you give examples? The reason I'm asking is we probably want to let our users distinguish between data fetcher internal errors and query errors.

Reply:
I added it initially as a abortion exception for failed max complexity and then it was added as a exception for FieldValidation. So I think it belongs to `GraphQLQueryException`. 

The naming is my fault and should be handled in another PR.